rudder-lang/src/parser/error.rs

```diff
@@ -166,30 +184,103 @@ where
     E: Fn() -> PErrorKind<PInput<'src>>,
 {
     move |input| {
-        let x = f(input);
-        match x {
+        match f(input) {
             // a non nom error cannot be superseded
             // keep original context when possible
             Err(Err::Failure(err)) => match err.kind {
                 PErrorKind::Nom(_) => Err(Err::Failure(PError {
-                    context: err.context,
+                    context: None,
```

Comment:
why are you removing the original context ? 
it may contain useful data, especially if you forget or don't need to override it